C/cpu.c

```diff
@@ -0,0 +1,191 @@
+#include "simplicity/cpu.h"
+
+#if defined(__SHA__) && defined(__SSE4_1__) && (defined(__x86_64__) || defined(__amd64__) || defined(__i386__))
+#define SHA_NI
+#endif
+
+#ifdef SHA_NI
+
+#include <cpuid.h>
```

Comment:
3563f344d22b59ce27af5c07ff7998259d742699: If any header is undefined on the target architecture then there will be an error, right? It seems that x86_64, amd64 and i386 all support cpuid, etc.

Reply:
`cpuid.h` is provided by gcc and clang.  It may just fail to exist on other compilers.